### PR TITLE
Fs cloning traitlet

### PR DESCRIPTION
**Pull request by mpacer**
This addresses the first piece of the #154 plan, setting up the configuration and validation.

### CODE REVIEW

**Comment by mpacer** [comment]
@chyzzqo2 This is ready to review.

File: bookstore/bookstore_config.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class BookstoreSettings(LoggingConfigurable):
         config=True
     )
     published_prefix = Unicode("published", help="Prefix for published notebooks").tag(config=True)
-    enable_cloning = Bool(True, help="Enable cloning.").tag(config=True)
+    enable_s3_cloning = Bool(True, help="Enable cloning.").tag(config=True)
```
Comment (willingc) [inline bookstore/bookstore_config.py]: Let's loop back and add these config options to the docstring.

File: bookstore/bookstore_config.py [hunk 2]
```diff
@@ -62,6 +62,10 @@ class BookstoreSettings(LoggingConfigurable):
         16, help="Maximum number of threads for the threadpool allocated for S3 read/writes"
     ).tag(config=True)
 
+    fs_cloning_basedir = Unicode(
+        "", help=("Base directory used for relative paths when cloning from the file system")
```
Comment (willingc) [inline bookstore/bookstore_config.py]:
```suggestion
        "", help=("Base directory used for relative paths when cloning from the local file system")
```

File: bookstore/tests/test_bookstore_config.py [hunk 3]
```diff
@@ -46,31 +49,47 @@ def test_validate_bookstore_endpoint():
...
+        "s3_clone_valid": True,
+        "fs_clone_valid": False,
     }
     settings = BookstoreSettings(s3_bucket="A_bucket")
     assert validate_bookstore(settings) == expected
 
 
 def test_disable_cloning():
-    """Tests that all bookstore features validate with an s3_bucket."""
+    """Tests that cloning can be disabled when s3_bucket is provided."""
```
Comment (willingc) [inline bookstore/tests/test_bookstore_config.py]: Perhaps """Tests that cloning from s3_bucket can be disabled."""

File: bookstore/tests/test_bookstore_config.py [hunk 4]
```diff
@@ -46,31 +49,47 @@ def test_validate_bookstore_endpoint():
...
-        "clone_valid": False,
+        "s3_clone_valid": False,
+        "fs_clone_valid": False,
     }
-    settings = BookstoreSettings(s3_bucket="A_bucket", enable_cloning=False)
+    settings = BookstoreSettings(s3_bucket="A_bucket", enable_s3_cloning=False)
+    assert validate_bookstore(settings) == expected
+
+
+def test_enable_fscloning():
```
Comment (willingc) [inline bookstore/tests/test_bookstore_config.py]:
```suggestion
def test_enable_fs_cloning():
```

File: bookstore/bookstore_config.py (hunk 1 above)
Comment (mpacer) [reply bookstore/bookstore_config.py]: 👍

**Pull request merged**